tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,252 @@
+   DeleteSpaces(histExpression);
+
+   std::string histCut;
+   if (cutPos != std::string::npos)
+      histCut = line.substr(cutPos+kCutIntr.size());
+
+   auto check = fHists.insert(std::make_pair((const std::string&)histName,
+                                             std::make_pair(histExpression, histCut)));
+   if (!check.second)
+      return "Name of the histogram already existing";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case with the input file
+///
+/// \param[in] kFile name of the input file that has to be read
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& file): fInputName(file)
+{}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] name name of the tree
+/// \param[in] expressions what is showed in the histograms
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output, const std::vector<std::string>& inputFiles,
+                                 const std::string& name, std::vector<std::string> expressions):
+   fInputFiles(inputFiles), fOutputFile(output), fTreeName(name)
+{
+      for (std::string expr: expressions) {
+        std::string errMessage = HandleExpressionConfig(expr);
+         if (!errMessage.empty())
+            throw std::runtime_error(errMessage + " in " +  expr);
+      }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return true if the input arguments create the output file correctly
+
+bool TSimpleAnalysis::Analyze()
+{
+   TChain chain(fTreeName.c_str());
+   for (const std::string& inputfile: fInputFiles)
+      chain.Add(inputfile.c_str());
+   TFile ofile(fOutputFile.c_str(), "RECREATE");
+
+   for (const auto &histo : fHists) {
+      chain.Draw((histo.second.first + ">>" + histo.first).c_str(), histo.second.second.c_str(), "goff");
+      TH1F *ptrHisto = (TH1F*)gDirectory->Get(histo.first.c_str());
+      ptrHisto->Write();
+      if (ptrHisto == nullptr)
+         return false;
+   }
+   return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Return false if not a tree name
+///
+/// param[in] line line read from the input file
+
+static bool HandleTreeNameConfig(const std::string& line)
+{
+   return line.find("=") == std::string::npos;
+}
+
+
+////////////////////////////////////////////////////////////////////////////////
+/// Skip subsequent empty lines in a string and returns the number of the
+/// current line of the input file
+///
+/// param[in] numbLine number of the input file line
+
+std::string TSimpleAnalysis::SkipSubsequentEmptyLines(int& numbLine)
+{
+   std::string notEmptyLine;
+
+   while (getline(fIn,notEmptyLine) && DeleteSpaces(notEmptyLine)
```

Comment:
Create a new `TSimpleAnalysis::GetLine()` that calls `getline()` and `DeleteSpaces()` returns true if all is good.
`DeleteSpaces()` can then be void as it should be.